File: portfolio/projekte/migrate.py
import json
import random
import string

def get_id():
    return ''.join(random.choices(string.ascii_lowercase + string.digits, k=9))

def migrate_design(file_path):
    with open(file_path, 'r') as f:
        data = json.load(f)
        
    for p in data:
        blocks = []
        if 'images' in p:
            gallery_items = []
            for img in p['images']:
                if img.get('type') == 'pdf':
                    blocks.append({
                        'id': img.get('id', get_id()),
                        'type': 'pdf',
                        'title': img.get('title', ''),
                        'imageUrl': img.get('imageUrl', ''),
                        'pdfUrl': img.get('pdfUrl', '')
                    })
                elif img.get('imageUrl'):
                    gallery_items.append({'imageUrl': img.get('imageUrl'), 'title': img.get('title', '')})
                    
            if gallery_items:
                blocks.append({
                    'id': get_id(),
                    'type': 'gallery',
                    'title': 'Galerie',
                    'items': gallery_items
                })
            
            p['blocks'] = blocks
            del p['images']
            
    with open(file_path, 'w') as f:
        json.dump(data, f, indent=4, ensure_ascii=False)

def migrate_video(file_path):
    with open(file_path, 'r') as f:
        data = json.load(f)
        
    for p in data:
        blocks = []
        
        if 'pdf' in p and p['pdf']:
            pdf = p['pdf']
            blocks.append({
                'id': get_id(),
                'type': 'pdf',
                'title': pdf.get('title', 'Dokumentation'),
                'pdfUrl': pdf.get('link', ''),
                'imageUrl': pdf.get('image', '')
            })
            
        if 'videos' in p and p['videos']:
            for v in p['videos']:
                blocks.append({
                    'id': get_id(),
                    'type': 'youtube',
                    'title': v.get('title', ''),
                    'videoId': v.get('youtubeId', ''),
                    'tags': ', '.join(v.get('tags', [])) if isinstance(v.get('tags'), list) else ''
                })
                
        p['blocks'] = blocks
        if 'pdf' in p: del p['pdf']
        if 'videos' in p: del p['videos']
        
    with open(file_path, 'w') as f:
        json.dump(data, f, indent=4, ensure_ascii=False)

migrate_video('/Users/noeplain/Developer/noe-plain.github.io/portfolio/projekte/video-projects.json')
print('Migrated video-projects.json')
migrate_design('/Users/noeplain/Developer/noe-plain.github.io/portfolio/projekte/designs.json')
print('Migrated designs.json')
migrate_design('/Users/noeplain/Developer/noe-plain.github.io/portfolio/projekte/illustrations.json')
print('Migrated illustrations.json')

File: portfolio/projekte/migrate.js
const fs = require('fs');

function migrateDesign(file) {
    let data = JSON.parse(fs.readFileSync(file, 'utf8'));
    data = data.map(project => {
        let blocks = [];
        if (project.images) {
            let galleryItems = [];
            project.images.forEach(img => {
                if (img.type === 'pdf') {
                    blocks.push({
                        id: img.id || Math.random().toString(36).substr(2, 9),
                        type: 'pdf',
                        title: img.title,
                        imageUrl: img.imageUrl,
                        pdfUrl: img.pdfUrl
                    });
                } else if (img.imageUrl) {
                    galleryItems.push({ imageUrl: img.imageUrl, title: img.title });
                }
            });
            if (galleryItems.length > 0) {
                blocks.push({
                    id: Math.random().toString(36).substr(2, 9),
                    type: 'gallery',
                    title: 'Galerie',
                    items: galleryItems
                });
            }
        }
        
        project.blocks = blocks;
        delete project.images;
        return project;
    });
    fs.writeFileSync(file, JSON.stringify(data, null, 4), 'utf8');
}

function migrateVideo(file) {
    let data = JSON.parse(fs.readFileSync(file, 'utf8'));
    data = data.map(project => {
        let blocks = [];
        
        if (project.pdf) {
             blocks.push({
                 id: Math.random().toString(36).substr(2, 9),
                 type: 'pdf',
                 title: project.pdf.title || "Dokumentation",
                 pdfUrl: project.pdf.link,
                 imageUrl: project.pdf.image || ""
             });
        }

        if (project.videos) {
            project.videos.forEach(v => {
                blocks.push({
                    id: Math.random().toString(36).substr(2, 9),
                    type: 'youtube',
                    title: v.title,
                    videoId: v.youtubeId,
                    tags: v.tags ? v.tags.join(', ') : ''
                });
            });
        }
        
        project.blocks = blocks;
        delete project.pdf;
        delete project.videos;
        return project;
    });
    fs.writeFileSync(file, JSON.stringify(data, null, 4), 'utf8');
}

try {
    migrateVideo('/Users/noeplain/Developer/noe-plain.github.io/portfolio/projekte/video-projects.json');
    console.log('Migrated video-projects.json');
    migrateDesign('/Users/noeplain/Developer/noe-plain.github.io/portfolio/projekte/designs.json');
    console.log('Migrated designs.json');
    migrateDesign('/Users/noeplain/Developer/noe-plain.github.io/portfolio/projekte/illustrations.json');
    console.log('Migrated illustrations.json');
} catch (e) {
    console.error(e);
}
